Port AutoRecoveryDialog to wxDataViewListCtrl

leio opened the pull request:
Porting AutoRecoveryDialog to wxDataViewListCtrl (instead of wxListCtrl) gives us 2 main benefits:

- We stop using wxListCtrl, which is considered legacy, with wxDataViewCtrl and co being its replacement.
  - Unlike wxListCtrl, wxDataViewCtrl (and their convenience helpers like wxDataViewListCtrl used here) has native implementation for wxGTK and wxMac too. wxMSW looks a bit less "native" implementation, but they aren't exactly really native in either case, but it might be worth looking at how it is on Windows afterwards too.
- wxDataViewCtrl and co exist in wx3.0 already with full toggle column support, while wxListCtrl added a limited checkbox first column support only in 3.1.x, with maintainers noting even there that wxListCtrl is really sort of deprecated, but (paraphrased) "if you've done the work, why not get it in then).

In summary, this means we get a native list control in the crash dialog for GTK and Mac, while lowering the wxWidgets requirement for this dialog in the process.

I haven't compile tested this PR in isolation on my machine yet (due to it requiring 3.1), but it works with 3.0 when the rest of in-progress stuff in #300 is included.

I invite behaviour and other testing help, especially compared to what was there before with wxListCtrl. I nuked some code that we might want to bring back in keyboard navigation stuff, and there was some weird select-all stuff in the wxListCtrl variant, I believe.

Based on that I can tweak it some more to be better.

<details>
<summary>Checklist</summary>

- [x] I have signed off my commits using `-s` or `Signed-off-by`\* (See: [Contributing § DCO](https://github.com/tenacityteam/tenacity/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))
- [ ] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code\*
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving\*
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"\*

\* indicates required

</details>

solfisher commented:
Builds and launches fine on Alpine Linux with 3.1, but it only shows one recoverable project even though there are two.

leio commented:
I was able to see even 5 recoverable projects if I kept "crashing" them unsaved. I don't think there's any difference in the code that picks which rows actually exist, other than some iteration changes?

solfisher commented:
I feel dumb -- all of the crashes are listed, but the list is only one row high. I need to scroll to see everything else.

leio commented:
That's actually a UX bug to be fixed by me then, there should be plenty of height to just have a higher area for the listing. Albeit for me there was no scrollbar needed until 7 items or so

solfisher commented:
macOS using 3.1 also works, it has enough space for 2 and a bit rows. I'm not sure why the box size wouldn't be uniform.

leio commented:
I'm passing wxDefaultSize for the size, I'll see about forcing something reasonable. How are things compared to the old thing before this PR?

solfisher commented:
The old list is too wide and has a horizontal scrollbar, even though there's no content hidden -- just blank space. This isn't the case with the new one.

Other than that, I haven't noticed anything different.

leio commented:
I added back a minimum height of 120, as that seemed to be the best compromise between gtk2 and gtk3 for me, so I hope it gives good results for MSW and Mac too. It's what was used by the old control too, except together with forcing some width and a concrete height, instead of a minimum.

The problem with this is that the numbers mean largely different things between gtk2 and gtk3 HiDPI, it's like for gtk3 it's multplied, while on gtk2 it's not. I tried 200 and 150, but that made it on gtk3 HiDPI a bit too high. 120 seems to result in at least 2 shown on gtk2 and 4 shown on gtk3 HiDPI, whereas as only minimum is requested, not a given height forced, it fits about 6-7 lines, but allows the dialog to be shrunk down to showing 4 at once.

So if this is looking OK on Mac and MSW as well, it should theoretically at least be the same kind of height than the old thing, just sometimes maybe a bit higher, as only minimum is forced.

I also dropped an outdated TODO that I checked over and that call is needed for a bit better behaviour for keyboard handling.

solfisher commented:
Working on Alpine with 3.1, shows 4 lines -- I'll test on macOS later today.

solfisher commented:
Also works for me on macOS with 3.1. Once somebody tests this on Windows or FreeBSD, I'll approve.

leio commented:
This should make no difference on FreeBSD compared to Linux, other than different gtk version or similar effects, as opposed to platform difference.

solfisher reviewed (approved):
Works fine on Windows, too.
